tests/datastore_pymongo/test_pymongo.py

```diff
@@ -0,0 +1,219 @@
+import pymongo
+import pytest
+import sqlite3
+
+from testing_support.fixtures import (validate_transaction_metrics,
+    validate_transaction_errors, validate_database_duration)
+from testing_support.db_settings import mongodb_settings
+
+from newrelic.api.background_task import background_task
+
+
+DB_SETTINGS = mongodb_settings()[0]
+MONGODB_HOST = DB_SETTINGS["host"]
+MONGODB_PORT = DB_SETTINGS["port"]
+
+
+def _exercise_mongo(db):
+    db.my_collection.save({"x": 10})
+    db.my_collection.save({"x": 8})
+    db.my_collection.save({"x": 11})
+    db.my_collection.find_one()
+
+    for item in db.my_collection.find():
+        item["x"]
+
+    db.my_collection.create_index("x")
+
+    for item in db.my_collection.find().sort("x", pymongo.ASCENDING):
+        item["x"]
+
+    [item["x"] for item in db.my_collection.find().limit(2).skip(1)]
+
+    if pymongo.version_tuple >= (3, 0):
+        db.my_collection.initialize_unordered_bulk_op()
+        db.my_collection.initialize_ordered_bulk_op()
+        db.my_collection.bulk_write([pymongo.InsertOne({'x': 1})])
+        db.my_collection.insert_one({'x': 300})
+        db.my_collection.insert_many([{'x': 1} for i in range(20, 25)])
+        db.my_collection.replace_one({'x': 1}, {'x': 2})
+        db.my_collection.update_one({'x': 1}, {'$inc': {'x': 3}})
+        db.my_collection.update_many({'x': 1}, {'$inc': {'x': 3}})
+        db.my_collection.delete_one({'x': 4})
+        db.my_collection.delete_many({'x': 4})
```

Comment:
It looks like there are destructive operations executing on the same DB. Is this safe for concurrent executions via tox -p auto?

Reply:
I hadn't encountered any issues with concurrency but reading the docs it definitely looks like it could fail. I'll get it changed to be parameterized.